worker/src/lib/clients/S3Client.ts

```diff
@@ -70,4 +140,97 @@ export class S3Client {
       return { data: null, error: error?.message };
     }
   }
+
+  async uploadBase64ToS3(
+    base64Data: string,
+    assetType: string,
+    requestId: string,
+    orgId: string
+  ): Promise<string> {
+    const buffer = Buffer.from(base64Data, "base64");
+    const fileExtension = this.getFileExtension(assetType);
+    const uploadUrl = this.getRequestResponseImageUrl(
+      requestId,
+      orgId,
+      fileExtension
+    );
+
+    await this.uploadToS3(uploadUrl, buffer, assetType);
+    return uploadUrl;
+  }
+
+  async uploadImageToS3(
+    imageUrl: string,
+    requestId: string,
+    orgId: string
+  ): Promise<string> {
+    const response = await fetch(imageUrl);
+    if (!response.ok) {
+      console.error(`Failed to download image: ${response.statusText}`);
+      return "";
+    }
+    const blob = await response.blob();
+    const fileExtension =
+      this.getFileExtensionFromBlob(blob) ||
+      this.getFileExtensionFromUrl(imageUrl);
+    const uploadUrl = this.getRequestResponseImageUrl(
+      requestId,
+      orgId,
+      fileExtension ?? "jpg"
+    );
+
+    await this.uploadToS3(uploadUrl, await blob.arrayBuffer(), blob.type);
+    return uploadUrl;
+  }
+
+  private getFileExtensionFromBlob(blob: Blob) {
```

Comment:
I think all of these getFileExtension methods should go into a utils class.

ideally S3Client just wraps S3 commands and the business logic lives elsewhere.